NDTensors/ext/NDTensorscuTENSORExt/contract.jl

```diff
@@ -12,9 +12,12 @@ function NDTensors.contract!(
   α::Number=one(Bool),
   β::Number=zero(Bool),
 )
-  cuR = CuTensor(array(unexpose(R)), collect(labelsR))
-  cuT1 = CuTensor(array(unexpose(T1)), collect(labelsT1))
-  cuT2 = CuTensor(array(unexpose(T2)), collect(labelsT2))
+  ## TODO for now a hack to get cuTENSOR working with blocksparse
+  R = unexpose(R)
+  cuR = CuTensor(copy(array(R)), collect(labelsR))
+  cuT1 = CuTensor(copy(array(unexpose(T1))), collect(labelsT1))
+  cuT2 = CuTensor(copy(array(unexpose(T2))), collect(labelsT2))
   cuTENSOR.mul!(cuR, cuT1, cuT2, α, β)
+  copyto!(data(R), cuR.data)
```

Comment:
Can we use:
```julia
copyto!(array(R), cuR.data)
```
here? It would be nice to avoid using `data(...)` which is pretty low level. Also we could use broadcasting syntax:
```julia
array(R) .= cuR.data
```
which is a bit more compact.

Finally, is there a more official API for converting a `CuTensor` to a `CuArray` besides `t.data`? Maybe `convert(CuArray, t)`? It seems like we shouldn't be using `t.data` since that seems like an internal interface but maybe there isn't anything better.

Reply:
That is a good recommendation. This primary push was a validation that all the unittests would work as I was seeing on my workstation. I will update the code to use a more official function for grabbing the cuTENSOR data. Thanks!

Reply:
Makes sense, thanks for letting me know this is for testing at first. There may not be an official function for unwrapping the `CuArray` besides `t.data`, but worth looking into (and maybe making a PR with a definition like `Base.convert(::Type{CuArray}, t::CuTensor) = t.data`).

Reply:
I updated the code as we discussed and am closing this comment.